pandas/core/array_algos/masked_reductions.py

```diff
@@ -45,3 +45,42 @@ def sum(
             return np.sum(values[~mask])
         else:
             return np.sum(values, where=~mask)
+
+
+def _minmax(func):
+    def reduction(values: np.ndarray, mask: np.ndarray, skipna: bool = True):
+        """
+        Reduction for 1D masked array.
+
+        Parameters
+        ----------
+        values : np.ndarray
+            Numpy array with the values (can be of any dtype that support the
+            operation).
+        mask : np.ndarray
+            Boolean numpy array (True values indicate missing values).
+        skipna : bool, default True
+            Whether to skip NA.
+        """
+        if not skipna:
+            if mask.any():
+                return libmissing.NA
+            else:
+                if values.size:
+                    return func(values)
+                else:
+                    # min/max with empty array raise in numpy, pandas returns NA
+                    return libmissing.NA
+        else:
+            subset = values[~mask]
+            if subset.size:
+                return func(values[~mask])
+            else:
+                # min/max with empty array raise in numpy, pandas returns NA
+                return libmissing.NA
+
+    return reduction
+
+
+min = _minmax(np.min)
+max = _minmax(np.max)
```

Comment:
since mypy is a static checker, min and max will resolve to Any.

something to consider is that maybe could do something like

```
def min(values: np.ndarray, mask: np.ndarray, skipna: bool = True):
    return _reduction(np.min, values=values, mask=mask, skipna=skipna)


def max(values: np.ndarray, mask: np.ndarray, skipna: bool = True):
    return _reduction(np.max, values=values, mask=mask, skipna=skipna)
```

and remove  _minmax and move func into reduction signature. 

This issue occurs elsewhere in the codebase, so this is not a blocker, but may give more confidence in the refactors.

alternatively, would using functools.partial be clearer. (newer versions on mypy might support this better)

Reply:
OK, adapted as you suggested (it gives an extra function call in runtime (instead of only at import time), but that should be negligible)